sktime/utils/validation/series.py

```diff
@@ -83,13 +90,20 @@ def check_time_index(index, allow_empty=False):
     if isinstance(index, np.ndarray):
         index = pd.Index(index)
 
-    # We here check for type equality because isinstance does not work reliably
-    # because index types inherit from each other.
-    if not type(index) in VALID_INDEX_TYPES:
-        raise NotImplementedError(
-            f"{type(index)} is not supported, use "
-            f"one of {VALID_INDEX_TYPES} instead."
-        )
+    if index_type:
+        if type(index) is not index_type:
+            raise ValueError(
+                f"{type(index)} is not supported, use "
+                f"type {index_type} instead."
+            )
+    else:
+        # We here check for type equality because isinstance does not
+        # work reliably because index types inherit from each other.
+        if not type(index) in VALID_INDEX_TYPES:
```

Comment:
I'd check for `VALID_INDEX_TYPES` first and regardless of `index_type` is given, we wouldn't want to support other index types not in VALID_INDEX_TYPES, but be more restrictive in some cases.

Reply:
done